### PR TITLE
UICHKIN-287: UI tests replacement with RTL/Jest for view CheckIn

### PR DESCRIPTION
**Pull request by artem-blazhko**
## Purpose
Cover CheckiIn.js by jest.RTL tests.

## Refs
[UICHKIN-287](https://issues.folio.org/browse/UICHKIN-287)

### CODE REVIEW

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]
I know Checkin.test.js is a new added test file.

All Buttons in Action menu are conditionally rendered, looks like the conditional scenarios are not addressed. Could you please add them?

File: src/Checkin.test.js [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+        ...basicProps,
+        scannedItems: [
+          {
+            item: {},
+          }
+        ]
+      };
+
+      beforeEach(() => {
+        isDCBItem.mockReturnValue(true);
```
Comment (Terala-Priyanka) [inline src/Checkin.test.js:643]: Although you are mocking the method isDCBItem, and the test are passing successfully, the pre-conditions established are not correct for DCB scenarios..

Just to give you some background on DCB wrt to Checkin app - 
_An item is called as a DCB item when its instanceid and holdingsRecordId are some specific values - refs - https://github.com/folio-org/ui-checkin/blob/c496d3adb3bf0a89d9f584a4b83dc9d9e8ecd1a5/src/consts.js#L66, https://github.com/folio-org/ui-checkin/blob/c496d3adb3bf0a89d9f584a4b83dc9d9e8ecd1a5/src/consts.js#L67, https://github.com/folio-org/ui-checkin/blob/c496d3adb3bf0a89d9f584a4b83dc9d9e8ecd1a5/src/util.js#L174
When it is DCB Item, only  Item Details button is hidden. Refs - https://github.com/folio-org/ui-checkin/pull/608
DCB User-(Also called virtual user)
A user is called as dcb user when his user type is dcb and lastname is DcbSystem
For a DCB user, only New fee/fine button is hidden. Refs - https://github.com/folio-org/ui-checkin/pull/606
The other buttons or menu items in action menu are rendered as is.
Your tests in test suite named - "When all data for buttons visibility are not set" are contradicting and not establishing the correct pre conditions for the tests._

It is probably a good idea to have separate test suites that match the scenario - for instance, have a separate test suite for DCB item and DCB user

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]

File: src/Checkin.test.js (hunk 1 above)
Comment (artem-blazhko) [reply src/Checkin.test.js:643]: You left a comment about external isDCBItem method which is placed in another file and it is mocked in my test. I just mocked it and that's it. It does not matter how it is implemented and what DCB item or DCB user means I just need a few different values (true or false) to test all my tests scenarios.
When isDCBItem returns true, item details button should not be visible, when it returns false button should be visible. That's all what I need.
We do not need to rely on inner implementation of isDCBItem and isDcbUser methods inside the test.

File: src/Checkin.test.js [hunk 2]
```diff
@@ -617,12 +621,12 @@ describe('CheckIn', () => {
       });
     });
 
-    describe('When all data for buttons visibility are not set', () => {
+    describe('When there is a loan on DCP item', () => {
```
Comment (Terala-Priyanka) [inline src/Checkin.test.js:624]: Please rename DCP to DCB

**Review by Terala-Priyanka — APPROVED** [review]

**Pull request merged**